sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs

```diff
@@ -64,7 +62,7 @@ public AzureMonitorTransmitter(AzureMonitorExporterOptions options, TokenCredent
             {
                 try
                 {
-                    var storageDirectory = options.StorageDirectory ?? StorageHelper.GetDefaultStorageDirectory();
+                    var storageDirectory = options.StorageDirectory ?? StorageHelper.GetDefaultStorageDirectory() ?? throw new InvalidOperationException("Unable to determine offline storage directory.");
```

Comment:
behavior change?

Reply:
it looks like that, but not really a change.

`StorageHelper.GetDefaultStorageDirectory` can fail and return null.
That then causes the `new FileBlobProvider` constructor to throw.

Reply:
but we were not throwing before right?

Reply:
This throw does not crash the process. Catch block logs into event source. If this is acting as a blocker for this PR better to remove the changes and track it on an issue.

Reply:
My bad - I was not reading it properly.